x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/CsvSpecReader.java

```diff
@@ -142,6 +142,22 @@ public void adjustExpectedWarnings(Function<String, String> updater) {
         public List<Pattern> expectedWarningsRegex() {
             return expectedWarningsRegex;
         }
+
+        /**
+         * How should we assert the warnings returned by ESQL.
+         */
+        public AssertWarnings assertWarnings(boolean deduplicateExact) {
+            if (expectedWarnings.isEmpty() == false) {
+                if (deduplicateExact) {
+                    return new AssertWarnings.DeduplicatedStrings(expectedWarnings);
+                }
+                return new AssertWarnings.ExactStrings(expectedWarnings);
```

Comment:
Nit: the ternary operator would make this and below more legible to me, but it's fine this way too.